### PR TITLE
Storage module

### PR DESCRIPTION
**Pull request by cif**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -26,7 +25,7 @@ jobs:
 
       - name: Build & Test
         run: |
-          npm ci
+          npm i
```
Comment (tifrel) [inline .github/workflows/publish.yml:28]: But shouldn't it be `ci`?

File: packages/storage/package.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  "files": [
+    "lib"
+  ],
+  "keywords": [],
+  "author": "",
+  "license": "MIT",
+  "dependencies": {
+    "@bundlr-network/client": "^0.7.7",
+    "near-api-js": "^0.44.2",
+    "superagent": "^8.0.3"
```
Comment (tifrel) [inline packages/storage/package.json:21]: According to Ruben, we are using isomorphic-unfetch as our HTTP request polyfill

File: packages/storage/src/uploads.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  } catch (err: unknown) {
+    const httpError = err as HttpError;
+    console.error(
+      `Uploading file to arweave failed: ${httpError.status} ${httpError.response.text}`,
+    );
+    throw err;
+  }
+};
+
+// export const uploadBuffer = async (
```
Comment (tifrel) [inline packages/storage/src/uploads.ts:59]: Dead code?

**Review by SurgeCode — APPROVED** [review]

File: secret.config.js [hunk 4]
```diff
@@ -0,0 +1 @@
+process.env.MINTBASE_API_KEY=e886efa80e627f165ebbc7b4d942c172
```
Comment (tifrel) [inline secret.config.js:1]: Should this be public?

**Review by tifrel — COMMENTED** [review]
Changes in `/package-lock.json` are massive. Also changed `packages/rpc/package-lock.json` with no apparent reason. Shouldn't the latter already exist on `alpha`?

File: packages/storage/src/uploads.ts (hunk 3 above)
Comment (cif) [reply packages/storage/src/uploads.ts:59]: This is front @microchipgnu first pass at bring your own key / client side Bundlr direct vs the upload service

File: secret.config.js (hunk 4 above)
Comment (cif) [reply secret.config.js:1]: This key itself isn't sensitive but may be in the future, seems like this PR might now be missing commits :/

File: .github/workflows/publish.yml (hunk 1 above)
Comment (cif) [reply .github/workflows/publish.yml:28]: Thought changed default version in nvm but guess that doesn't work at system level.  Either one here is fine really, when we were dealing with the node version issues this was attempt to workaround

File: secret.config.js (hunk 4 above)
Comment (cif) [reply secret.config.js:1]: Ah, nm we are just using .env via dotconfig now

**Pull request merged**